xarray/tests/test_parallelcompat.py

```diff
@@ -147,6 +157,35 @@ def register_dummy_chunkmanager(monkeypatch):
     yield
 
 
+class TestPassThroughNonRegisteredChunkedArrays:
+    """
+    Check that types which implement .chunks and .rechunk are still dispatched to for these methods, even if they are not registered via a ChunkManager.
+
+    Basically regression tests for GH issue #8733.
+
+    Notice we specifically do not use the register_dummy_chunkmanager fixture in these tests.
+    """
+
+    def test_chunks(self) -> None:
+        dummy_arr = DummyChunkedArray(shape=(9,), chunks=((3,),))
+        na: NamedArray = NamedArray(data=dummy_arr, dims=["x"])
+        assert na.chunks == ((3,),)
+        assert na.chunksizes == {"x": (3,)}
+
+    def test_rechunk(self) -> None:
+        dummy_arr = DummyChunkedArray(shape=(4,), chunks=((1,),))
+        na: NamedArray = NamedArray(data=dummy_arr, dims=["x"])
+        rechunked_na = na.chunk(chunks={"x": (2,)})
+        assert isinstance(rechunked_na.data, DummyChunkedArray)
+        assert rechunked_na.data.chunks == ((2,),)
+        assert rechunked_na.chunksizes == {"x": (2,)}
+
+    def test_computation(self) -> None:
+        dummy_arr = DummyChunkedArray(shape=(4,), chunks=((1,),))
+        na: NamedArray = NamedArray(data=dummy_arr, dims=["x"])
+        na.mean()
```

Comment:
Not entirely sure what the intended behaviour should be here. This test tests what happens if you try to compute an array that has `.chunks` but is not registered via any chunkmanager. 

In virtualizarr's case this situation should just raise immediately because `ManifestArray`s are not computable, so from virtualizarr's PoV it doesn't really matter what happens here.

@hmaarrfk what is the preferred behaviour for your chunked arrays?

I guess if it does attempt to pass computation through here that could cause issues when computing on a cubed array with cubed-xarray not installed... (That scenario can't happen for dask because the equivalent DaskManager (i.e. dask-xarray) is effectively bundled inside xarray.)

Reply:
The users of my array (the rest of our team) feels like all these should do something.

It might make things REALLY slow, but I feel like `mean` should compute.... Your chunked array should know how best to compute it for itself. How should it compute intermediate results? In what order should it go through the array.

Reply:
> It might make things REALLY slow, but I feel like mean should compute....

Yes I agree.

> How should it compute intermediate results? In what order should it go through the array.

I'm not sure I understand you here.

> I guess if it does attempt to pass computation through here that could cause issues when computing on a cubed array with cubed-xarray not installed...

Thinking about this more I don't think it's a big deal. Not recognising a chunked array type will just mean that xarray falls back to calling numpy functions on it (e.g. `da.mean()` will call `np.mean(arr)`), which will call `__array__` on the underlying type, coercing it to numpy, and in the case of cubed arrays, simply eagerly computing it.

Reply:
> cubed arrays, simply eagerly computing it.

maybe cubed would help us in our lazy arrays.

Reply:
Me personally that is

Reply:
> Thinking about this more I don't think it's a big deal.

You also kinda have to go out of your way to even created an xarray-wrapped cubed array without cubed-xarray installed, because you can only use `open_dataset` and `.chunk` to get cubed arrays if you have cubed-xarray installed.

> maybe cubed would help us in our lazy arrays.

Maybe! The [`cubed.Plan`](https://cubed-dev.github.io/cubed/computation.html#plan) model is super nice.